src/libutil/util.cc

```diff
@@ -678,31 +678,36 @@ void drainFD(int fd, Sink & sink, bool block)
 {
     int saved;
 
+    if (!block) {
+        saved = fcntl(fd, F_GETFL);
+        if (fcntl(fd, F_SETFL, saved | O_NONBLOCK) == -1)
+            throw SysError("making file descriptor non-blocking");
+    }
+
     Finally finally([&]() {
         if (!block) {
+            #pragma GCC diagnostic push
+            // gcc detects `saved` as maybe uninitialized
```

Comment:
Easier to just do `int saved = 0` to shup up the warning.

Reply:
changed to do that instead.